examples/railroad-network.flix

```diff
@@ -0,0 +1,79 @@
+// Define a function that decides whether pairs of railroad stations in `links` are safely connected.
+//
+// Given two railroad stations a and b, if they are only connected through some station c,
+// then they are unsafely connected. Because if c fails there is no way to get from a to b.
+def safeConnects(src: n, links: f[(n, n)]): {safe::List[n], unsafe::List[n]} with Foldable[f], Boxable[n] =
+    let db = inject links into Link;
+    let pr = #{
+        // Directly linked stations.
+        Linked(a, b) :- Link(a, b).
+        Linked(a, b) :- Link(b, a).
+
+        // Indirectly linked stations.
+        Connected(a, b) :- Linked(a, b).
+        Connected(a, b) :- Connected(a, c), Linked(c, b).
+
+        // Station x that is a cutpoint between a and b.
+        Cutpoint(x, a, b) :- Connected(a, b), Station(x), not Circumvent(x, a, b).
+
+        // ???
+        Circumvent(x, a, b) :- Linked(a, b), if x != a, Station(x), if x != b.
+        Circumvent(x, a, b) :- Circumvent(x, a, c), Circumvent(x, c, b).
+
+        // Check the cutpoint is neither station a nor b.
+        HasICutPoint(a, b) :- Cutpoint(x, a, b), if x != a, if x != b.
+        // Stations are safely connected if there is no cutpoints.
+        SafelyConnected(a, b) :- Connected(a, b), not HasICutPoint(a, b).
+        // The set of stations not safely connected are unsafely connected.
+        UnsafelyConnected(a, b) :- Station(a), Station(b), not SafelyConnected(a, b).
+        // All stations.
+        Station(x) :- Linked(x, _).
```

Comment:
We also need `Station(src).` as an additional fact. I might not actually be needed but I think its clearer if we just include it.